### DISCUSSION TITLE
Firebase auth

### DISCUSSION THREAD

**Discussion by gr-qft** [vercel/next.js#18036] [Help]
I'm following along this example 

https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication

For some reason the function 

```js
export const mapUserData = user => {
  console.log({ user });
  const { uid, email, xa } = user;
  return {
    id: uid,
    email,
    token: xa
  };
};
```

causes the following message 

```
{
  error: FirebaseAuthError: Decoding Firebase ID token failed. Make sure you passed the entire string JWT which represents an ID token. See https://firebase.google.com/docs/auth/admin/verify-id-tokens for details on how to retrieve an ID token...
}
```

I checked and `xa` is always false. I did a console log and saw there was `ya` also, and using `ya` removes the problem (so the app is working). Could someone help me with understanding what's going on: what are `xa` and `ya`, and why using `ya` works? 

Thanks!

**Comment by futantan** [comment]
Same issue!

**Comment by futantan** [comment]
I changed it to `user.getIdToken()` since I didn't find any document about `xa` or `ya`

**Comment by gr-qft** [comment]
I `console.log(user.getIdToken())` and it's an object with a bunch of weird keys a,b,c... and so I'm not sure now how it should be used.

**Reply by futantan** [reply]
`user.getIdToken()` return a Promise, try `console.log(await user.getIdToken())`
BTW, the demo updated to `xa || ya` https://github.com/vercel/next.js/blob/canary/examples/with-firebase-authentication/utils/auth/mapUserData.js#L6

**Comment by gr-qft** [comment]
Ok I was able to get a token out of it but using that token lead to the food failing to fetch again.
